video2dataset/dataloader/audio_decode.py

```diff
@@ -0,0 +1,30 @@
+"""Audio Decoders"""
+import torchaudio
+import io
+
+
+def set_backend(extension):
+    """Set torchaudio backend for different extensions (soundfile doesn't support M4A and MP3)"""
+    if extension in ["wav", "flac"]:
+        torchaudio.set_audio_backend("soundfile")
+    else:
+        torchaudio.set_audio_backend("sox_io")
+
+
+class AudioDecoder:
+    """Basic audio decoder that converts audio into torch tensors"""
+
+    def __init__(self, sample_rate=None, num_channels=None):
+        self.sample_rate = sample_rate
+        self.num_channels = num_channels
+
+    def __call__(self, key, data):
+        extension = key.split(".")[-1]
+        set_backend(extension)
```

Comment:
is this slow? because if it is maybe it would be better to just set this once based on info passed in the dataloader (kind of like video_key). But if its not slow then its fine just leave this

Reply:
it seems to slightly slow things down so I changed that